tests.md

```diff
@@ -0,0 +1,219 @@
+# Testing of Terraform Modules Explained
+
+## Unit Tests Do Not Use Apply
```

Comment:
Changed the title. The titles are used as url fragments, so they better be stable. Please double-check all titles.

Reply:
Checked, consistent.